### PR TITLE
[W5.11][T12-2]Cheng Xianhao

### PR DESCRIPTION
**Pull request by xhxh96**

### CODE REVIEW

File: src/seedu/addressbook/parser/Parser.java [hunk 1]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (chokxy) [inline src/seedu/addressbook/parser/Parser.java:14]: Should have listed out all the dependencies instead of the "*"

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -72,6 +72,11 @@ Examples:
 Shows a list of all persons, along with their non-private details, in the address book. +
 Format: `list`
 
+== Listing all persons : `sort`
```
Comment (marlenekoh) [inline docs/UserGuide.adoc:75]: good job updating documentation

File: src/seedu/addressbook/commands/SortCommand.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import java.util.List;
+
+/**
+ * Sort all persons in the lexicographical order.
```
Comment (marlenekoh) [inline src/seedu/addressbook/commands/SortCommand.java:7]: nice that you've added javadoc comments

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 4]
```diff
@@ -1,11 +1,6 @@
 package seedu.addressbook.data.person;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
+import java.util.*;
```
Comment (marlenekoh) [inline src/seedu/addressbook/data/person/UniquePersonList.java:3]: Agree, please change your intellij java import settings to show only relevant imports

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 5]
```diff
@@ -81,6 +76,18 @@ public UniquePersonList(UniquePersonList source) {
...
     }
 
+    /**
+     * Returns an unmodifiable java List view with sorted elements cast as immutable {@link ReadOnlyPerson}s.
+     * For use with other methods/libraries.
+     * Any changes to the internal list/elements are immediately visible in the returned list.
+     */
+    public List<ReadOnlyPerson> sortedImmutableListView() {
+        List<Person> sortedInternalList = internalList;
+        sortedInternalList.sort(Comparator.comparing(p -> p.getName().fullName));
```
Comment (marlenekoh) [inline src/seedu/addressbook/data/person/UniquePersonList.java:86]: good

File: src/seedu/addressbook/parser/Parser.java [hunk 6]
```diff
@@ -99,6 +89,9 @@ public Command parseCommand(String userInput) {
         case ExitCommand.COMMAND_WORD:
             return new ExitCommand();
 
+        case SortCommand.COMMAND_WORD:
+            return new SortCommand();
+
         case HelpCommand.COMMAND_WORD: // Fallthrough
```
Comment (marlenekoh) [inline src/seedu/addressbook/parser/Parser.java:95]: Remember to go into Help command to add message usage for sort command (:

File: test/expected.txt [hunk 7]
```diff
@@ -293,6 +293,32 @@
 || Enter command: || [Command entered:  clear]
 || Address book has been cleared!
 || ===================================================
+|| Enter command: || [Command entered:  add Adam Brown p/111111 e/adam@gmail.com a/111, alpha street]
```
Comment (marlenekoh) [inline test/expected.txt:296]: Good job updating relevant tests

**Review by marlenekoh — COMMENTED** [review]
@xhxh96 good job for this enhancement and updating docs and test cases (: 

Please close this PR after reading review comments.

**Comment by xhxh96** [comment]
Noted with thanks.

**Pull request closed without merging**